components/SingleClassView/Attendance/AttendanceComponent.tsx

```diff
@@ -33,13 +39,22 @@ export const AttendanceComponent: React.FC<AttendanceComponentProps> = ({ classI
       );
       if (sessions.length == 0) {
         setAttendance([]);
+        setStudentAttendance([]);
         setLoading(false);
         return <p>This date has no sessions</p>;
       }
       setSessionID(sessions[0].sessionId);
       const sessionId = sessions[0].sessionId;
-      const attendances = await api.getAttendanceFromSessionID(sessionId, classId);
-      setAttendance(attendances);
+      if (user){
+        if (user.role == "Teacher" || user.role == "Admin"){
+          const attendances = await api.getAttendanceFromSessionID(sessionId, classId);
+          setAttendance(attendances);
+        }
+        else{
+          const studentAttendance = await api.getSingleUserAttendanceFromSessionID(user.id, sessionId, classId);
+          setStudentAttendance(studentAttendance);
```

Comment:
since parents should be able to access this page for their students, how about we pass an id in via an optional prop. If its set, we get the information of that user, else we get it from the logged in user id